Merge frame splitting into sw dev

sander-willems-bruker opened the pull request.

sander-willems-bruker commented:
@jspaezp Not sure how I accomplished the merge without retaining your git history, no really a github expert to be honest. If you know how to fix that, please feel free to do so:). Note that this PR also implemented the domain inversion functions you require (https://github.com/MannLabs/timsrust/pull/23/commits/e53b95807433ef1a6059f3282272d0c2932f4692)

jspaezp commented:
I can tidy those prints in the next PR if that is OK with you! lmk if you want a cascading PR onto this one or to use dev as a new base once we merge this one (to add the default options + splitting in terms of 1/k0 instead of scans). and feel free to merge whenever.

sander-willems-bruker commented:
I removed print statements and merged into develop. The default are reasonably set at the moment (I introduced builders for SpectrumReaders and PrecursorReaders, whereas I resctructed code so FrameReader is untouched as our changes where only at the SpectrumLevel). Future implementations to split in terms of 1/k0 instead of scans are best made on a new branch of develop. We also will need to make a new Sage PR of course to propagate the config info to timsrust from there. I am happy to look into that, good excercise for me as I havent worked much with serde yet:)
